src/circuit_json.rs

```diff
@@ -144,6 +173,29 @@ pub enum OpBox {
         n_o: u32,
         exp: ClassicalExp,
     },
+    /// A user-defined multiplexor specified by a map from bitstrings to Operations.
+    MultiplexorBox {
+        id: BoxID,
+        op_map: Vec<(Vec<bool>, Operation)>,
+    },
+    /// A user-defined multiplexed rotation gate specified by a map from
+    /// bitstrings to Operations.
+    MultiplexedRotationBox {
+        id: BoxID,
+        op_map: Vec<(Vec<bool>, Operation)>,
+    },
+    /// A user-defined multiplexed rotation gate specified by a map from
+    /// bitstrings to Operations.
+    MultiplexedU2Box {
+        id: BoxID,
+        op_map: Vec<(Vec<bool>, Operation)>,
+        #[serde(default = "default_impl_diag")]
+        impl_diag: bool,
+    },
+}
+
+fn default_impl_diag() -> bool {
```

Comment:
not very happy about adding this function

Reply:
Yeah, there isn't really a way around it :/